tests/foreman/cli/test_variables.py

```diff
@@ -0,0 +1,1120 @@
+        3. Duplicate values in YAML output if any are displayed.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_matcher_merge_default(self):
+        """Test empty default value in merged values.
+
+        @id: 98f8fe63-d125-4d27-a15a-2550c9e5f0ff
+
+        @steps:
+
+        1. Create variable with some default value.
+        2. Create first matcher for attribute fqdn with valid details.
+        3. Create second matcher for other attribute with valid details.
+        Note - The fqdn/host should have this attribute.
+        4. Create more matchers for some more attributes if any.
+        Note - The fqdn/host should have this attributes.
+        5. Set --merge-overrides to true.
+        6. Set --merge-default to true.
+        7. Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all
+        the associated matchers.
+        2.  The YAML output doesn't have the empty default value of variable.
+        3.  Duplicate values in YAML output if any are displayed.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_avoid_duplicate(self):
+        """Merge the values of all the associated matchers, remove duplicates.
+
+        @id: 75fc514f-70dd-4cc1-8069-221e9edda89a
+
+        @steps:
+
+        1.  Create variable with type array and value.
+        2.  Create first matcher for attribute fqdn with some value.
+        3.  Create second matcher for other attribute with
+        same value as fqdn matcher.
+        Note - The fqdn/host should have this attribute.
+        4.  Set --merge-overrides to true.
+        5.  Set --merge-default to true.
+        6.  Set --avoid -duplicates' to true.
+        8.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all
+        the associated matchers.
+        2.  The YAML output has the default value of variable.
+        3.  Duplicate values in YAML output are removed / not displayed.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_matcher_avoid_duplicate(self):
+        """Duplicates not removed as they were not really present.
+
+        @id: 050c7cef-eed6-4a61-b567-371f398647a2
+
+        @steps:
+
+        1.  Create variable with type array and value.
+        2.  Create first matcher for attribute fqdn with some value.
+        3.  Create second matcher for other attribute with other value
+        than fqdn matcher and default value.
+        Note - The fqdn/host should have this attribute.
+        4.  Set --merge-overrides to true.
+        5.  Set --merge-default to true.
+        6.  Set --avoid -duplicates' to true.
+        7.  Go to YAML output of associated host.
```

Comment:
We don't have yaml output in CLI, even if you specify --output-format=yaml, it will not be the same as we have for UI